arch/arc/core/mpu/arc_mpu.c

```diff
@@ -460,17 +460,6 @@ void arc_core_mpu_configure_user_context(struct k_thread *thread)
 	}
 
 	arc_core_mpu_configure(THREAD_STACK_USER_REGION, base, size);
-
-#if defined(CONFIG_APP_SHARED_MEM) && CONFIG_ARC_MPU_VER == 3
```

Comment:
Need to replace CONFIG_APP_SHARED_MEM with CONFIG_USERSPACE and that will properly fix this.

Reply:
yes, CONFIG_APP_SHARED_MEM should be replaced with CONFIG_USERSPACE

Reply:
Fixed